### PR TITLE
Added Reciplan schema

### PR DESCRIPTION
**Pull request by mpadil18**
Removed difficulty category from all recipes table
Added dish_type table to track all dish types by recipe id.
Considering changing name in script from recipath to plan

### CODE REVIEW

File: backend/recipath-schema.sql [hunk 1]
```diff
@@ -8,3 +8,53 @@ CREATE TABLE users (
...
+    date            TIMESTAMP NOT NULL DEFAULT NOW(),
+    FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+);
+
+CREATE TABLE all_recipes (
+    id              SERIAL PRIMARY KEY,
+    title           TEXT NOT NULL,
+    category        TEXT NOT NULL,
+    image_url       TEXT NOT NULL,
+    prep_time       INT NOT NULL,
```
Comment (awang2330) [inline backend/recipath-schema.sql]: should this be integer?

File: backend/recipath-schema.sql [hunk 2]
```diff
@@ -8,3 +8,53 @@ CREATE TABLE users (
...
+    region          TEXT NOT NULL,
+    short_bio       TEXT NOT NULL,
+    fav_flavors     TEXT NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+);
+
+CREATE TABLE saved_recipes (
+    id              SERIAL PRIMARY KEY,
+    user_id         INTEGER NOT NULL,
+    recipe_id       INTEGER NOT NULL,
```
Comment (awang2330) [inline backend/recipath-schema.sql]: for all recipe ids do we need a foreign key to link to recipe table

**Comment by mpadil18** [comment]
Updated saved_recipes and saved_meal_plans to include foreign key references to all recipes table

**Review by awang2330 — APPROVED** [review]
looks good

**Pull request merged**